helix-term/src/commands.rs

```diff
@@ -1791,6 +1792,29 @@ fn search_selection(cx: &mut Context) {
     cx.editor.set_status(msg);
 }
 
+fn make_search_word_bounded(cx: &mut Context) {
+    let regex = match cx.editor.registers.last('/') {
+        Some(regex) => regex,
+        None => return,
+    };
```

Comment:
I think the way that `search_selection` currently works (by pushing a new value to the end of the register) is a bit odd and might be refactored in #4504 

Instead I think this function should pull out the values of the register (`cx.editor.registers.read(register)`, add word boundaries to all of them and then write them back `cx.editor.registers.write(register, new_values)`. That will make it harder to say what the new regex is in the statusline but I think we can say something simpler, maybe `"word boundaries added to register '{}'"` is sufficient